EclipseLink uses wrong type, String, instead of List of String when retuning a subset of attributes as a record

I attempted to use a Java record,
```
public record RomanNumeral(
                String name,
                String romanNumeral,
                ArrayList<String> romanNumeralSymbols) {}
```

and the JPQL,
```
SELECT NEW test.jakarta.data.web.RomanNumeral(name, romanNumeral, romanNumeralSymbols) FROM Prime WHERE numberId <= ?1 ORDER BY name
```

to retrieve a subset of attributes of the entity,
```
<entity class="test.jakarta.data.web.Prime">
  <table name="Prime"/>
  <attributes>
   ...
   <basic name="name">
   </basic>
   <id name="numberId">
    <column nullable="false"/>
   </id>
   <basic name="romanNumeral">
   </basic>
   <element-collection name="romanNumeralSymbols" fetch="EAGER">
   </element-collection>
   ...
  </attributes>
 </entity>
```

```
public class Prime {
    ...
    public String name;
    public long numberId;
    public String romanNumeral;
    public ArrayList<String> romanNumeralSymbols;
    ...
```

When running the JPQL query, EclipseLink tried to use a single String value equal to the first element of the ArrayList<String> collection as the value of romanNumeralSymbols instead of a value that is the collection, resulting in the error:

```
< invoke Primes.romanNumeralsLessThanEq Exit  
                                                                                                               Local Exception Stack: 
Exception [EclipseLink-6137] (Eclipse Persistence Services - 5.0.0-B05.v202412161836-e4245e29fa3355029f10c42ea5546c73683dd504): org.eclipse.persistence.exceptions.QueryException
Exception Description: An Exception was thrown while executing a ReportQuery with a constructor expression: java.lang.NoSuchMethodException: test.jakarta.data.web.RomanNumeral.<init>(java.lang.String,java.lang.String,java.lang.String)
Internal Exception: java.lang.NoSuchMethodException: test.jakarta.data.web.RomanNumeral.<init>(java.lang.String,java.lang.String,java.lang.String)
Query: ReportQuery(referenceClass=Prime jpql="SELECT NEW test.jakarta.data.web.RomanNumeral(name, romanNumeral, romanNumeralSymbols) FROM Prime WHERE numberId <= ?1 ORDER BY name")
	at org.eclipse.persistence.exceptions.QueryException.exceptionWhileUsingConstructorExpression(QueryException.java:537)
	at org.eclipse.persistence.queries.ConstructorReportItem.initialize(ConstructorReportItem.java:208)
	at org.eclipse.persistence.queries.ReportQuery.prepare(ReportQuery.java:1153)
	at org.eclipse.persistence.queries.DatabaseQuery.checkPrepare(DatabaseQuery.java:696)
	at org.eclipse.persistence.queries.ObjectLevelReadQuery.checkPrepare(ObjectLevelReadQuery.java:1031)
	at org.eclipse.persistence.queries.DatabaseQuery.checkPrepare(DatabaseQuery.java:645)
	at org.eclipse.persistence.internal.jpa.EJBQueryImpl.buildEJBQLDatabaseQuery(EJBQueryImpl.java:196)
	at org.eclipse.persistence.internal.jpa.EJBQueryImpl.buildEJBQLDatabaseQuery(EJBQueryImpl.java:120)
	at org.eclipse.persistence.internal.jpa.EJBQueryImpl.<init>(EJBQueryImpl.java:107)
	at org.eclipse.persistence.internal.jpa.EJBQueryImpl.<init>(EJBQueryImpl.java:91)
	at org.eclipse.persistence.internal.jpa.EntityManagerImpl.createQuery(EntityManagerImpl.java:1846)
	at org.eclipse.persistence.internal.jpa.EntityManagerImpl.createQuery(EntityManagerImpl.java:1869)
	at io.openliberty.data.internal.persistence.RepositoryImpl.invoke(RepositoryImpl.java:734)
	at jdk.proxy14/jdk.proxy14.$Proxy55.romanNumeralsLessThanEq(Unknown Source)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.jboss.weld.bean.proxy.InterceptedProxyMethodHandler.invoke(InterceptedProxyMethodHandler.java:52)
	at test.jakarta.data.web.Primes$Proxy$_$$_WeldInterceptedProxy1.romanNumeralsLessThanEq(Unknown Source)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.jboss.weld.bean.proxy.AbstractBeanInstance.invoke(AbstractBeanInstance.java:38)
	at org.jboss.weld.bean.proxy.ProxyMethodHandler.invoke(ProxyMethodHandler.java:109)
	at test.jakarta.data.web.Primes$1501238887$Proxy$_$$_WeldClientProxy.romanNumeralsLessThanEq(Unknown Source)
	at test.jakarta.data.web.DataTestServlet.testQueryWithRecordResult(DataTestServlet.java:4372)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at componenttest.app.FATServlet.doGet(FATServlet.java:74)
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:633)
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:723)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1266)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:754)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:451)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1362)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1078)
	at com.ibm.ws.webcontainer.servlet.CacheServletWrapper.handleRequest(CacheServletWrapper.java:77)
	at com.ibm.ws.webcontainer40.servlet.CacheServletWrapper40.handleRequest(CacheServletWrapper40.java:87)
	at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:978)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:293)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1260)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:476)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:435)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:569)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:503)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:363)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.ready(HttpInboundLink.java:330)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.sendToDiscriminators(NewConnectionInitialReadCallback.java:169)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.complete(NewConnectionInitialReadCallback.java:77)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:516)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:586)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:970)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1059)
	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:298)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.NoSuchMethodException: test.jakarta.data.web.RomanNumeral.<init>(java.lang.String,java.lang.String,java.lang.String)
	at java.base/java.lang.Class.getConstructor0(Class.java:3761)
	at java.base/java.lang.Class.getConstructor(Class.java:2442)
	at org.eclipse.persistence.internal.security.PrivilegedAccessHelper.getConstructorFor(PrivilegedAccessHelper.java:402)
	at org.eclipse.persistence.queries.ConstructorReportItem.initialize(ConstructorReportItem.java:204)
	... 53 more
```

--- a/dev/io.openliberty.data.internal_fat/test-applications/DataTestApp/src/test/jakarta/data/web/RomanNumeral.java
+++ b/dev/io.openliberty.data.internal_fat/test-applications/DataTestApp/src/test/jakarta/data/web/RomanNumeral.java
@@ -10,17 +10,15 @@
  *******************************************************************************/
 package test.jakarta.data.web;
 
-import java.util.ArrayList;
-
 /**
  * A record to use as a result type for selecting attributes of the Prime entity.
+ * EclipseLink decided to disallow collection attributes such as
+ * ArrayList<String> romanNumeralSymbols
+ * so it is omitted from from this record.
  */
 public record RomanNumeral(
                 String name,
-                String romanNumeral
-// , TODO enable once EclipseLink bug #30501 is fixed
-// ArrayList<String> romanNumeralSymbols
-) {
+                String romanNumeral) {
 
     /**
      * Format in an easy way for tests to compare results.
@@ -30,9 +28,9 @@ public record RomanNumeral(
         StringBuilder s = new StringBuilder();
         s.append(name).append(" ");
         s.append(romanNumeral).append(" ( ");
+        // If collection attributes are ever allowed, the following could become
+        // for (String symbol : romanNumeralSymbols)
         for (char symbol : romanNumeral.toCharArray())
-        // TODO replace the above with the following once EclipseLink bug #30501 is fixed
-        //for (String symbol : romanNumeralSymbols)
             s.append(symbol).append(' ');
         s.append(")");
         return s.toString();

--- a/dev/io.openliberty.data.internal_fat_jpa/test-applications/DataJPATestApp/src/test/jakarta/data/jpa/web/DataJPATestServlet.java
+++ b/dev/io.openliberty.data.internal_fat_jpa/test-applications/DataJPATestApp/src/test/jakarta/data/jpa/web/DataJPATestServlet.java
@@ -2014,8 +2014,7 @@ public class DataJPATestServlet extends FATServlet {
     /**
      * Verify that JPQL can be used to EXTRACT the DATE from a LocalDateTime.
      */
-    // TODO enable once EclipseLink bug #31802 is fixed
-    //@Test
+    @Test
     public void testExtractDate() {
         rebates.reset();
 
@@ -2138,8 +2137,7 @@ public class DataJPATestServlet extends FATServlet {
     /**
      * Verify that JPQL can be used to EXTRACT the TIME from a LocalDateTime.
      */
-    // TODO enable once EclipseLink bug #31802 is fixed
-    //@Test
+    @Test
     public void testExtractTime() {
         rebates.reset();
 

--- a/dev/io.openliberty.data.internal_fat_jpa/test-applications/DataJPATestApp/src/test/jakarta/data/jpa/web/DataJPATestServlet.java
+++ b/dev/io.openliberty.data.internal_fat_jpa/test-applications/DataJPATestApp/src/test/jakarta/data/jpa/web/DataJPATestServlet.java
@@ -1332,20 +1332,27 @@ public class DataJPATestServlet extends FATServlet {
                                              .map(a -> a.houseNumber + " " + a.streetName)
                                              .collect(Collectors.toList()));
 
-        // TODO Enable once EclipseLink bug #31558 is fixed:
-        // List<ShippingAddress> found = shippingAddresses
-        //                .findByStreetAddressRecipientInfoNotEmpty();
+        List<ShippingAddress> found = shippingAddresses
+                        .findByStreetAddressRecipientInfoNotEmpty();
+        ShippingAddress a = null;
+        for (ShippingAddress s : found) {
+            if (a1.id.equals(s.id))
+                a = s;
+        }
+        // TODO Replace above for loop with the following once EclipseLink bug #31559 is fixed
         // assertEquals(1, found.size());
-        // ShippingAddress a = found.get(0);
-        // assertEquals(a1.id, a.id);
-        // assertEquals(a1.city, a.city);
-        // assertEquals(a1.state, a.state);
-        // assertEquals(a1.zipCode, a.zipCode);
-        // assertEquals(a1.streetAddress.houseNumber, a.streetAddress.houseNumber);
-        // assertEquals(a1.streetAddress.streetName, a.streetAddress.streetName);
-        // assertEquals(a1.streetAddress.recipientInfo, a.streetAddress.recipientInfo);
+        // a = found.get(0);
+        assertEquals(a1.id, a.id);
+        assertEquals(a1.city, a.city);
+        assertEquals(a1.state, a.state);
+        assertEquals(a1.zipCode, a.zipCode);
+        assertEquals(a1.streetAddress.houseNumber, a.streetAddress.houseNumber);
+        assertEquals(a1.streetAddress.streetName, a.streetAddress.streetName);
+        assertEquals(a1.streetAddress.recipientInfo, a.streetAddress.recipientInfo);
 
-        // assertEquals(3L, shippingAddresses.countByStreetAddressRecipientInfoEmpty());
+        long count = shippingAddresses.countByStreetAddressRecipientInfoEmpty();
+        // TODO Enable once EclipseLink bug #31559 is fixed:
+        // assertEquals(3L, count);
 
         // [EclipseLink-4002] Internal Exception: java.sql.SQLIntegrityConstraintViolationException:
         //                    DELETE on table 'SHIPPINGADDRESS' caused a violation of foreign key constraint 'SHPPNGSHPPNGDDRSSD' for key (1001)

--- a/dev/io.openliberty.data.internal_fat_jpa/test-applications/DataJPATestApp/src/test/jakarta/data/jpa/web/DataJPATestServlet.java
+++ b/dev/io.openliberty.data.internal_fat_jpa/test-applications/DataJPATestApp/src/test/jakarta/data/jpa/web/DataJPATestServlet.java
@@ -1446,13 +1446,15 @@ public class DataJPATestServlet extends FATServlet {
                                                  .map(t -> t.ssn)
                                                  .collect(Collectors.toList()));
 
-        // TODO enable once issue #31558 and
-        // TODO subsequently #32263 is fixed in EclipseLink
-        if (false)
-            assertIterableEquals(List.of(789007890L),
-                                 taxpayers.findByBankAccountsNotEmpty()
-                                                 .map(t -> t.ssn)
-                                                 .collect(Collectors.toList()));
+        assertEquals(List.of(123001230L,
+                             234002340L,
+                             345003450L,
+                             456004560L,
+                             567005670L,
+                             678006780L),
+                     taxpayers.findByBankAccountsNotEmpty()
+                                     .map(t -> t.ssn)
+                                     .collect(Collectors.toList()));
 
         taxpayers.delete();
     }

